src/simulator.ts

```diff
@@ -86,8 +86,9 @@ export class Simulator {
     return this.processing || this.closeDuration > 0;
   }
 
-  public cancel() {
+  public async cancel() {
     this.cancelToken = true;
+    await this.delay(1000);
```

Comment:
why we need to delay?

Reply:
1. In the worst case, it takes 500ms to truly stop the progress.
2. The cancel button will change to `disabled` style as soon as user clicks it, and comes back to normal style when the `set` function returns.

Based on the above 2 situation, if there is no delay, this function will quickly set cancelToken=true, and return, so user may feel confused:
1. when they see the button becomes disabled and comes back to normal in 500ms(if they are high sensitive enough to feel that in 500ms), but the process is not canceled.
2. So, a possible solution is to click the button more than once in 500ms(if they are capable of clicking more than once in 500ms), which may lead to more confusions.

When considering normal situations, it seems like that the delay is unnecessary. But by adding this, everyone wil see the button become disabled ---> and the progress truly canceled ---> and then the button come back normal ---> finally the Modal disappers.

So, no one will have any confusion.

The delay will not lead to any loss in performance, since it just make the button stay in `disabled` state for 1 second.